sycl/plugins/unified_runtime/ur/adapters/cuda/enqueue.cpp

```diff
@@ -393,9 +431,24 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
       UR_CHECK_ERROR(RetImplEvent->start());
     }
 
-    if (hQueue->getContext()->getDevice()->maxLocalMemSizeChosen()) {
+    // Once event has been started we can unlock MemoryMigrationMutex
+    for (auto &MemArg : hKernel->Args.MemObjArgs) {
+      // Telling the ur_mem_handle_t that it will need to wait on this kernel
+      // if it has been written to
```

Comment:
We actually don't care if we have a read -> write dep since if the write kernel is on the same queue, the normal dep management using streams will work. If the write kernel is on a different device then the read kernel can happen in parallel with the subsequent write, since the write kernel will be writing to an allocation on a different device and the read enqueue kernel already knows that the memory is in the correct state on its device.